src/scripting/lua_attributes.hpp

```diff
@@ -0,0 +1,162 @@
+/*
+	Copyright (C) 2009 - 2024
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+/// New attribute registration system, mainly for objects with a lot of attributes, like units
+/// Not used for GUI2 widgets, as they're even more complicated with a deep hierarchy.
+
+#pragma once
+
+struct lua_State;
+class t_string;
+class vconfig;
+
+#include "config.hpp"
+#include "variable_info.hpp"
+#include "map/location.hpp"
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+struct luaW_Registry {
```

Comment:
Some comments would be nice, especially on what the Boolean parameters of the callback functions do